backend/core/migrations/0005_email_unique_in_db.py

```diff
@@ -0,0 +1,62 @@
+import logging
+from django.db import migrations, models
+from django.db.models import Count, Window, F
+from django.db.models.functions.window import RowNumber
+
+
+def remove_duplicated_email(apps, schema_editor):
+    """
+    Find users with duplicated emails, and update email of the accounts associated
+    with the fewer comparisons (to preserve only 1 per original address).
+    Their 'email' value will be replaced with an invalid address derived
+    from the username to guarantee uniqueness in the database.
+    """
+    User = apps.get_model("core", "User")
+
+    duplicated_emails = (
+        User.objects.values("email").alias(count=Count("id")).filter(count__gte=2)
+    )
+    users_to_update = [
+        user
+        for user in User.objects.filter(email__in=duplicated_emails).annotate(
+            n_comparisons=Count("comparisons"),
+            rank_comparisons=Window(
+                expression=RowNumber(),
+                partition_by=[F("email")],
+                order_by=F("n_comparisons").desc(),
+            ),
+        )
+        if user.rank_comparisons > 1
+    ]
+
+    for u in users_to_update:
+        new_email = f"{u.username}@invalid"
+        logging.info(
+            'Updating email for user "%s", from "%s" to "%s"',
+            u.username,
+            u.email,
+            new_email,
+        )
```

Comment:
I would consider using a `logger.warning`, because the fact of creating two accounts with the same email is not recommended globally on websites.